src/semver.nim

```diff
@@ -0,0 +1,85 @@
+##
+## Copyright (c) 2023, Crash Override, Inc.
+##
+## This file is part of Chalk
+## (see https://crashoverride.com/docs/chalk)
+##
+import strutils, util
+
+# very simple semver implementation
+# it this does NOT handle full semver spec
+# it only handles basic dot-separated version format
+
+type Version* = ref object
+    parts: seq[int]
+
+proc parseVersion*(version: string): Version =
+  new result
+  result.parts = @[]
+  for i in version.strip(chars={'v', ',', '.'}).split('.'):
```

Comment:
This also relies on parseInt to throw an error for malformed stuff, which makes this highly error prone as a parser. If people accidentally pass in the wrong thing, they could get spurious versions, etc.

OR it could blow up on things it probably should accept, like if people do semver but w/ release candidates... we've all seen `v2.1.0-rc1` for instance. Personally I'd be much more precise and explicit when it comes to what is or is not allowed.